tests/php/test-class-amp-dom-document.php

```diff
@@ -0,0 +1,328 @@
+				'<html>' . $head . '<body><p>مرحبا بالعالم! Check out ‘this’ and “that” and—other things.</p></body>',
+				'<!DOCTYPE html><html>' . $head . '<body><p>مرحبا بالعالم! Check out ‘this’ and “that” and—other things.</p></body></html>',
+			],
+			'utf_8_encoding_guessed_via_content'       => [
+				'',
+				'<p>مرحبا بالعالم! Check out ‘this’ and “that” and—other things.</p>',
+				'<!DOCTYPE html><html>' . $head . '<body><p>مرحبا بالعالم! Check out ‘this’ and “that” and—other things.</p></body></html>',
+			],
+			'iso_8859_1_encoding_predefined'           => [
+				'iso-8859-1',
+				utf8_decode( '<!DOCTYPE html><html><head></head><body><p>ÄÖÜ</p></body></html>' ),
+				'<!DOCTYPE html><html>' . $head . '<body><p>ÄÖÜ</p></body></html>',
+			],
+			'iso_8859_1_encoding_guessed_via_charset'  => [
+				'',
+				utf8_decode( '<!DOCTYPE html><html><head><meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1" /></head><body><p>ÄÖÜ</p></body></html>' ),
+				'<!DOCTYPE html><html>' . $head . '<body><p>ÄÖÜ</p></body></html>',
+			],
+			'iso_8859_1_encoding_guessed_via_content'  => [
+				'',
+				utf8_decode( '<!DOCTYPE html><html><body><p>ÄÖÜ</p></body></html>' ),
+				'<!DOCTYPE html><html>' . $head . '<body><p>ÄÖÜ</p></body></html>',
+			],
+			'raw_iso_8859_1'                           => [
+				'',
+				utf8_decode( 'ÄÖÜ' ),
+				'<!DOCTYPE html><html>' . $head . '<body>ÄÖÜ</body></html>',
+			],
+			// Make sure we correctly identify the ISO-8859 sub-charsets ("€" does not exist in ISO-8859-1).
+			'iso_8859_15_encoding_predefined'          => [
+				'iso-8859-1',
+				mb_convert_encoding( '<!DOCTYPE html><html><head></head><body><p>€</p></body></html>', 'ISO-8859-15', 'UTF-8' ),
+				'<!DOCTYPE html><html>' . $head . '<body><p>€</p></body></html>',
+			],
+			'iso_8859_15_encoding_guessed_via_charset' => [
+				'',
+				mb_convert_encoding( '<!DOCTYPE html><html><head><meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-15" /></head><body><p>€</p></body></html>', 'ISO-8859-15', 'UTF-8' ),
+				'<!DOCTYPE html><html>' . $head . '<body><p>€</p></body></html>',
+			],
+			'iso_8859_15_encoding_guessed_via_content' => [
+				'',
+				mb_convert_encoding( '<!DOCTYPE html><html><body><p>€</p></body></html>', 'ISO-8859-15', 'UTF-8' ),
+				'<!DOCTYPE html><html>' . $head . '<body><p>€</p></body></html>',
+			],
+			'raw_iso_8859_15'                          => [
+				'',
+				mb_convert_encoding( '€', 'ISO-8859-15', 'UTF-8' ),
+				'<!DOCTYPE html><html>' . $head . '<body>€</body></html>',
+			],
+		];
+	}
+
+	/**
+	 * Tests loading and saving the content via Amp\AmpWP\Dom\Document.
+	 *
+	 * @param string $charset  Charset to use.
+	 * @param string $source   Source content.
+	 * @param string $expected Expected target content.
+	 *
+	 * @dataProvider data_dom_document
+	 * @covers Document::loadHTML()
+	 * @covers Document::saveHTML()
+	 */
+	public function test_dom_document( $charset, $source, $expected ) {
```

Comment:
The `$charset` param is unused. Is that intentional?

Reply:
No, it's not, good catch!

Reply:
Resolved via https://github.com/ampproject/amp-wp/pull/3758/commits/c5415d2bbf9ae25f120c5de36d30a5c1068e0b80